tendermint-rs/src/abci/proof.rs

```diff
@@ -0,0 +1,51 @@
+//! ABCI Merkle proofs
+
+use crate::error::Error;
+use serde::{de::Error as _, Deserialize, Deserializer, Serialize, Serializer};
+use std::{
+    fmt::{self, Display},
+    str::FromStr,
+};
+use subtle_encoding::{Encoding, Hex};
+
+/// ABCI Merkle proofs
+#[derive(Clone, Debug, Eq, Hash, PartialEq)]
+pub struct Proof(Vec<u8>);
```

Comment:
@ebuchman this seems like it might have some overlap with the stuff you're working on? That said, is this an OK start?